management_api_app/api/routes/workspaces.py

```diff
@@ -55,6 +57,42 @@ async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo:
     return WorkspaceIdInResponse(workspaceId=workspace.id)
 
 
+@router.post("/workspaces/{workspace_id}/workspace-services", status_code=status.HTTP_202_ACCEPTED,
+             response_model=WorkspaceServiceIdInResponse, name=strings.API_CREATE_WORKSPACE_SERVICE)
+async def create_workspace_service(workspace_create: WorkspaceServiceInCreate,
+                                   workspace_service_repo: WorkspaceServiceRepository = Depends(
+                                       get_repository(WorkspaceServiceRepository)),
+                                   user: User = Depends(get_current_user),
+                                   workspace: Workspace = Depends(get_workspace_by_workspace_id_from_path)
+                                   ) -> WorkspaceServiceIdInResponse:
+    access_service = get_access_service()
+    if access_service.get_workspace_role(user, workspace) != WorkspaceRole.Owner:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=strings.ACCESS_USER_IS_NOT_OWNER)
+
+    try:
+        workspace_service = workspace_service_repo.create_workspace_service_item(workspace_create, workspace.id)
+    except (ValidationError, ValueError) as e:
+        logging.error(f"Failed create workspace service model instance: {e}")
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
+
+    try:
+        workspace_service_repo.save_workspace_service(workspace_service)
+    except Exception as e:
+        logging.error(f"Failed save workspace service instance in DB: {e}")
+        raise HTTPException(status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                            detail=strings.STATE_STORE_ENDPOINT_NOT_RESPONDING)
+
+    try:
+        await send_resource_request_message(workspace_service)
+    except Exception as e:
+        # TODO: Rollback DB change, issue #154
```

Comment:
I wasn't a big fan of this TODO in the workspaces API :) - rather than a TODO - can we delete the created item from the DB here?